pkg/cmd/roachtest/cluster.go

```diff
@@ -1585,6 +1585,59 @@ func (c *clusterImpl) FetchCores(ctx context.Context, l *logger.Logger) error {
 	})
 }
 
+// FetchPebbleCheckpoints fetches any Pebble checkpoints on the cluster. These
+// are typically generated by failed consistency checks, i.e. replica
+// divergence, and only contain data for the inconsistent ranges.
+func (c *clusterImpl) FetchPebbleCheckpoints(ctx context.Context, l *logger.Logger) error {
+	// Don't grab checkpoints on empty clusters or local runs.
+	if c.spec.NodeCount == 0 || c.IsLocal() {
+		return nil
+	}
+
+	// Checkpoints can be large. Bail out after 3 minutes.
+	return timeutil.RunWithTimeout(ctx, "checkpoints", 3*time.Minute, func(ctx context.Context) error {
+		numStores := 1
+		if ssds := c.spec.SSDs; ssds > 1 {
+			numStores = ssds
+		}
+		for storeIdx := 1; storeIdx <= numStores; storeIdx++ {
+			// Find any checkpoints.
+			//
+			// TODO(erikgrinaker): expose and use SyncedCluster.NodeDir(). We can't
+			// use the {store-dir} expansion which only refers to the first store.
+			storePath := fmt.Sprintf("/mnt/data%d/cockroach", storeIdx)
+			var checkpointNodes option.NodeListOption
+			results, err := c.RunWithDetails(ctx, l, c.All(), "sudo", "/bin/bash", "-c",
+				fmt.Sprintf("'find %s/auxiliary -maxdepth 1 -type d -name checkpoints'", storePath))
```

Comment:
>I believe this would eliminate the need to check for stdout

Yeah, that's better, thanks. I originally started out with a `find /mnt` here, to try to automagically handle all stores, so the code is a bit weird.

>Also curious: what error do we get without sudo?

Permission denied on `lost+found`, which only `root` has access to. But that was from a previous incarnation, it shouldn't be necessary now that we use `auxiliary` as the base. I'll remove it.